keps/sig-node/5067-pod-generation/README.md

```diff
@@ -0,0 +1,990 @@
+-->
+
+This proposal aims to allow the pod status to express which pod updates are 
+currently being reflected in the pod status. The idea is to leverage the 
+existing `metadata.Generation` field and add a new `status.observedGeneration` field to 
+the pod status. 
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+One of the motivations for this KEP comes from the existing ResizeStatus field. 
+In its original implementation, the ResizeStatus field was written to by both 
+the API server and the Kubelet, creating a race condition on updates. Removing 
+the Proposed state makes the Kubelet the only writer to that field, but leaves a
+gap in knowing whether the latest resize has been acknowledged by the Kubelet. 
+The changes proposed in this KEP resolve this gap.
+
+The ResizeStatus is used as an example here, but in practice this issue can be 
+generally found in any type of pod update.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+* Provide a general solution for the pod status to express which pod update is 
+currently being reflected.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+* Expand the set of mutable fields.
+* Populate `metadata.generation` and `status.ObservedGeneration` on static pods.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### Current behavior
+
+The pod [`metadata.generation`](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#metadata) field does exist today, and is 
+documented as "a sequence number representing a specific generation of the 
+desired state. Set by the system and monotonically increasing, per-resource." 
+
+Its current behavior in pods is:
+
+* Pod `metadata.generation` is not populated by default by the system.
+* The client can custom-set the `metadata.generation` on pod create.
+* `metadata.generation` cannot be updated by the client.
+* `metadata.generation` does not get incremented by the system when the podspec is updated.
+* `metadata.generation` does get incremented by the system upon deletion.
```

Comment:
To clarify, does "upon deletion" mean when the `DeletionTimestamp` is set? Trying to be precise since we're talking about API behaviors :)

Reply:
yup I updated the wording to specify `DeletionTimestamp`